### PR TITLE
Added ReadTheDocs support. Formatted documentation. Added docs.

### LINKED ISSUE #152: Add ReadTheDocs.io support. Update the documentation.
The current documentation is maintained in plain markdown files in the docs folder.  
To better organise and have a searchable documentation website it was decided to integrate with readthedocs.io  
The following tasks need to be done for the same:  

- Install and configure readthedocs.  
- Rearrange the docs folder  
- Format the documentation.  
- Integrate with readthedocs site.  
- Provide link on main website kubeedge.io

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
As per discussions with @skdwriting the documentation for the KubeEdge project will be maintained in the repository /docs folder. To make it easier for developers to read and search the documentation is now being hosted on readthedocs.io

**Which issue(s) this PR fixes**:
Fixes #152 

**Special notes for your reviewer**:
This is the [preview](https://kubeedge-anvith-rtd.readthedocs.io/en/rtd-sphinx/) of the documentation from my fork.

### CODE REVIEW

**Comment by anvithks** [comment]
@skdwriting @m1093782566 Please review and merge.

**Comment by anvithks** [comment]
@rohitsardesai83 Please review

**Comment by sids-b** [comment]
I was thinking if there is a possibility of moving the docs to website repo? If we decide to have cloud and edge part both in same repo, along with docs the repo will have lot of content.
@m1093782566 @rohitsardesai83 Thoughts ??

**Comment by anvithks** [comment]
@sids I discussed this with @skdwriting.
Developers will always have to visit the repository. 
The website will link to the Edge documentation and later the cloud documentation as well. 
When the cloud repository is online we will setup the readthedocs for the same.

**Comment by m1093782566** [comment]
@anvithks @skdwriting 

Do we need to merge this PR?

File: README.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 [![Go Report Card](https://goreportcard.com/badge/github.com/kubeedge/kubeedge)](https://goreportcard.com/report/github.com/kubeedge/kubeedge)
 [![LICENSE](https://img.shields.io/github/license/kubeedge/kubeedge.svg?style=flat-square)](https://github.com/kubeedge/kubeedge/blob/master/LICENSE)
 [![Releases](https://img.shields.io/github/release/kubeedge/kubeedge/all.svg?style=flat-square)](https://github.com/kubeedge/kubeedge/releases)
+[![Documentation Status](https://readthedocs.org/projects/kubeedge-anvith-sphinx/badge/?version=latest)](https://kubeedge-anvith-sphinx.readthedocs.io/en/latest/?badge=latest)
```
Comment (rohitsardesai83) [inline README.md]: this url should not have author name in it right ?

File: README_zh.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 [![Go Report Card](https://goreportcard.com/badge/github.com/kubeedge/kubeedge)](https://goreportcard.com/report/github.com/kubeedge/kubeedge)
 [![LICENSE](https://img.shields.io/github/license/kubeedge/kubeedge.svg?style=flat-square)](https://github.com/kubeedge/kubeedge/blob/master/LICENSE)
 [![Releases](https://img.shields.io/github/release/kubeedge/kubeedge/all.svg?style=flat-square)](https://github.com/kubeedge/kubeedge/releases)
+[![Documentation Status](https://readthedocs.org/projects/kubeedge-anvith-sphinx/badge/?version=latest)](https://kubeedge-anvith-sphinx.readthedocs.io/en/latest/?badge=latest)
```
Comment (rohitsardesai83) [inline README_zh.md]: same as previous comment.

File: docs/contribute-website.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+Thanks for taking the time to join our community and start contributing!
+
+## Quick guide to working with a GitHub repo
+
+Here's a quick guide to a fairly standard GitHub workflow. This section is handy
+for people who don't use git or GitHub often, and just need a quick guide to
+get going:
+
+1. Fork the kubeEdge/website repo:
```
Comment (rohitsardesai83) [inline docs/contribute-website.md]: kubeEdge -> kubeedge

File: docs/contribute-website.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    * Go to the [kubeEdge/website repo][kubeedge-website-repo] on GitHub.
```
Comment (rohitsardesai83) [inline docs/contribute-website.md]: kubeEdge -> kubeedge

File: docs/kubeedge.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+between cloud and edge. It also supports `MQTT` and allows developers to author customer logic
+and enable resource constraint devices communication at Edge. Kubeedge is *the open platform to enable Edge computing*.
+The advantages of Kubeedge include mainly:
+
+* **Edge Computing**
+
+     With business logic running at Edge, volumes of data can be secured & processed locally. It reduces the bandwidth
+     request between Edge and Cloud; increases the response speak; and protects customers' data privacy.
+
+* **Simplify development**
```
Comment (rohitsardesai83) [inline docs/kubeedge.md]: Simplify -> Simplified

File: docs/kubeedge.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+## The KubeEdge mission
+
+Our goal is to make an open platform to enable Edge computing, extending native containerized
+application orchestration capabilities to hosts at Edge. which built upon kubernetes and provides
+fundamental infrastructure support for network, app.deployment and metadata synchronization
+between cloud and edge. It also supports `MQTT` and allows developers to author customer logic
+and enable resource constraint devices communication at Edge. Kubeedge is *the open platform to enable Edge computing*.
+The advantages of Kubeedge include mainly:
+
+* **Edge Computing**
```
Comment (rohitsardesai83) [inline docs/kubeedge.md]: please see the readme.md in kubeedge/kubeedge. The description has been revised.

File: docs/setup.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+```shell
+yum install mosquitto
+```
+
+See [mosquitto official website](https://mosquitto.org/download/) for more information.
+
+## Build
+
+Clone kube-edge
```
Comment (rohitsardesai83) [inline docs/setup.md]: kube-edge  --> kubeedge

File: README.md (hunk 1 above)
Comment (anvithks) [reply README.md]: Will remove this for now. Will add the documentation status badge after we have the documentation live.

File: README_zh.md (hunk 2 above)
Comment (anvithks) [reply README_zh.md]: Same as before. I will remove this for now.

**Comment by anvithks** [comment]
> @anvithks @skdwriting
> 
> Do we need to merge this PR?

We are currently reviewing and fixing this PR. 
We will merge after it is approved.

File: README.md (hunk 1 above)
Comment (anvithks) [reply README.md]: Fixed this. Removed the line.

File: README_zh.md (hunk 2 above)
Comment (anvithks) [reply README_zh.md]: Fixed this. Removed the line.

File: docs/contribute-website.md (hunk 3 above)
Comment (anvithks) [reply docs/contribute-website.md]: Fixed this.

File: docs/contribute-website.md (hunk 4 above)
Comment (anvithks) [reply docs/contribute-website.md]: Fixed this.

File: docs/kubeedge.md (hunk 5 above)
Comment (anvithks) [reply docs/kubeedge.md]: Fixed this.

File: docs/kubeedge.md (hunk 6 above)
Comment (anvithks) [reply docs/kubeedge.md]: Fixed this. Referred the updated content and made changes accordingly.

File: docs/setup.md (hunk 7 above)
Comment (anvithks) [reply docs/setup.md]: Fixed this.

**Comment by anvithks** [comment]
> some review comments. also please validate if the readme.md in kubeedge/kubeedge is consistent and correct with the docs readme. thanks.

@rohitsardesai83  I have fixed the above review comments. Please check.  
@skdwriting Please review as well.

**Comment by sids-b** [comment]
@anvithks , can you please squash the commits ?

**Comment by anvithks** [comment]
> @anvithks , can you please squash the commits ?

Done. All the commits have been squashed into one.

**Comment by anvithks** [comment]
@sids-b sure. I will conduct a session with best practices of using markdown soon.

**Pull request merged**